### PR TITLE
g.extension: updating addon metadata file

### PR DESCRIPTION
**Pull request by anikaweinmann**
- for inofficial modules meta data informations are written to the addon modules.xml
- for the addons (e.g. i.sentinel.download, i.sentinel.import,  i.sentinel.mask, i.sentinel.preproc) in the meta-modules (i.sentinel) the meta data are written to the addon modules.xml

### CODE REVIEW

File: scripts/g.extension/g.extension.py [hunk 1]
```diff
@@ -685,9 +687,19 @@ def install_extension(source, url, xmlurl):
...
         # will provide the metadata file
-        if source == 'official':
+        if source == 'official' and len(installed_modules) <= len(mlist):
             grass.message(_("Updating addons metadata file..."))
             blist = install_extension_xml(xmlurl, mlist)
+        if source == 'official' and len(installed_modules) > len(mlist):
+            grass.message(_("Updating addons metadata file..."))
+            blist = install_inofficial_extension_xml(tmp_dir, installed_modules)
+        else:
+            grass.message(_("Updating inofficial addons metadata file..."))
```
Comment (neteler) [inline scripts/g.extension/g.extension.py]:
```suggestion
            grass.message(_("Updating private addons metadata file..."))
```

Comment (neteler) [reply scripts/g.extension/g.extension.py]: I am not sure which wording is better and more clear to the casual user..., you decide!

File: scripts/g.extension/g.extension.py [hunk 2]
```diff
@@ -912,6 +924,87 @@ def install_extension_xml(url, mlist):
     return bin_list
 
 
+def install_inofficial_extension_xml(url, mlist):
+    """Update XML files with metadata about installed modules and toolbox
+    of an inofficial addon
```
Comment (neteler) [inline scripts/g.extension/g.extension.py]:
```suggestion
    of a private addon
```

File: scripts/g.extension/g.extension.py [hunk 3]
```diff
@@ -912,6 +924,87 @@ def install_extension_xml(url, mlist):
...
+
+    """
+    # TODO toolbox
```
Comment (neteler) [inline scripts/g.extension/g.extension.py:932]: @wenzeslaus how to deal with "toolbox" here? Is that documented anywhere?

File: scripts/g.extension/g.extension.py (hunk 1 above)
Comment (NikosAlexandris) [reply scripts/g.extension/g.extension.py]: A suggestion:
- **Official** are add-ons accepted officialy by the core development team
- **Unofficial** are add-ons still in a repository owned by the GRASS GIS team, yet not meeting all requirements to be official (such as: lacking unit tests and testing overall, lacking clean documentation and references to original work, and similar).
- **Private** would mean add-ons that "I" can only see in my local computing environment.

**Comment by neteler** [comment]
Shall I now merge it in SVN? (in lack of answer we leave the "toolbox" part as a TODO...)

**Comment by neteler** [comment]
Merged into SVN along with some extra changes of http -> https
https://trac.osgeo.org/grass/changeset/74428
Thanks!

**Pull request closed without merging**